content/en/docs/guides/custom-name-resolution.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Custom Name Resolution
+description: >-
+  Explains standard name resolution, the custom name resolver interface,
+  and how to write an implementation.
+---
+
+### Overview
+
+Name resolution is fundamentally about service discovery. When sending a request
+to `grpc.io`, the client must determine the IP address of that service name.
+Name resolution is often thought to be the same as
+[DNS](https://www.ietf.org/rfc/rfc1035.txt). In practice however, DNS is usually
+augmented with extensions or completely replaced to enable name resolution.  For
+example, the `/etc/hosts` file on Posix systems is not strictly speaking part of
+DNS and instead stems from
+[RFC-952](https://datatracker.ietf.org/doc/html/rfc952).
+
+When making a request with a gRPC client, by default, DNS name resolution is
+used. However, various other name resolution mechanisms may be used:
+
+|Resolver|Example|Notes|
+|-|-|-|
+|DNS|`grpc.io:50051`|By default, DNS is assumed.|
+|DNS|`dns:///grpc.io:50051`|The extra slash is used to provide an authority|
+|Unix Domain Socket|`uds:///var/run/docker.sock`|
+|xDS|`xds:///wallet.grpcwallet.io`||
+|IPv4|`ipv4:198.51.100.123:50051`|Only supported in some languages|
+
+The triple slashes above (`///`) may look unfamiliar if you are used to the
+double slashes of HTTP, such as `https://grpc.io`. These _target strings_ follow
+the format for [RFC-3986](https://datatracker.ietf.org/doc/html/rfc3986) URIs.
+The string following the first two slashes and preceding the third (if there is
```

Comment:
The comparison of two slashes and three slashes seems too detailed. It might be better to put this information in a note so that only people who are interested can read it.